src/GeositeFramework/js/Export.js

```diff
@@ -34,71 +34,162 @@ require(['use!Geosite'],
+                return 'css/print-landscape.css';
+            } else {
+                return 'css/print-portrait.css';
+            }
+        }
 
-            // Setup a print-preview window for the user to select an extent and zoom level
-            // that will be persisted at print due to its fixed size.
+        function setupExport(context) {
+            var previewDeferred = $.Deferred(),
+                orientDeferred = $.Deferred(),
+                resizeDeferred = $.Deferred(),
+                postPrintAction = _.noop;
             
-            TINY.box.show({
-                animate: false,
-                html: $mapPrint[0].outerHTML,
-                boxid: 'export-print-preview-container',
-                width: 521,
-                height: 463,
-                fixed: true,
-                maskopacity: 40,
-                openjs: function () {
-                    var mapNode = $("#map-0").detach()[0];
-                    $("#export-print-preview-map").append(mapNode);
+            $('#export-button').on('click', function() {
+                var mapNode = $("#map-0").detach();
+                var exportMap = $("#export-print-preview-map");
 
-                    $('#export-button').on('click', function() {
-                        var $printSandbox = $('#map-print-sandbox'),
-                            previewDeferred = $.Deferred();
+                $('.print-sandbox-header h1').text($("#export-title").val());
+                exportMap.append(mapNode);
 
-                        $('.print-sandbox-header h1').text($("#export-title").val());
+                context.mapReadyDeferred.then(function () {
+                    exportMap.detach().appendTo($("#print-map-container"));
 
-                        mapReadyDeferred.then(function () {
-                            $("#export-print-preview-map").detach().appendTo($("#print-map-container"));
+                    var pageOrientation = $("[name='export-orientation']:checked").val();
+                    $('<link>', {
+                        rel: 'stylesheet',
+                        href: pageCssLink(pageOrientation),
+                        'class': '.print-orientation-css',
+                    }).appendTo('head');
+                    _.delay(orientDeferred.resolve, 500);
+
+                    $.when(orientDeferred).then(function () {
+                        context.map.width = parseFloat(exportMap.css("width"));
+                        context.map.height = parseFloat(exportMap.css("height"));
+                        context.map.resize(true);
+                        context.map.centerAt(context.mapDimensions.extent.getCenter());
+                        _.delay(resizeDeferred.resolve, 500);
+
+                        // the legend items are affected by the map resize, so
+                        // they are manipulated just before printing
+                        $.when(resizeDeferred).then(function() {
```

Comment:
I misunderstood what you meant by "nested." I think this could be updated to move totally separate parts out of blocks, I'll take a stab at it and push an update if it's a quick fix.